detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt

```diff
@@ -35,14 +39,20 @@ abstract class FormattingRule(config: Config, description: String) : Rule(config
 
     private lateinit var positionByOffset: (offset: Int) -> Pair<Int, Int>
     private lateinit var root: KtFile
+    private lateinit var originalPath: Path
 
     override fun visit(root: KtFile) {
-        this.root = root
-        positionByOffset = KtLintLineColCalculator.calculateLineColByOffset(root.text)
+        val fileCopy = KtPsiFactory(root.project).createPhysicalFile(root.name, root.modifiedText ?: root.text)
+
+        this.root = fileCopy
+        originalPath = root.absolutePath()
+        positionByOffset = KtLintLineColCalculator.calculateLineColByOffset(fileCopy.text)
 
         wrapping.beforeFirstNode(computeEditorConfigProperties())
-        root.node.visitASTNodes()
+        this.root.node.visitASTNodes()
         wrapping.afterLastNode()
+
+        root.modifiedText = this.root.text
```

Comment:
```suggestion
        if (this.root.modificationStamp > 0) {
            root.modifiedText = this.root.text
        }
```

Otherwise we will report that all the files were auto-formated.

Reply:
I've made the update here as suggested but I'll add the test in #7204